docs/baseband/vdif/index.rst

```diff
@@ -26,72 +26,59 @@ user-defined data.  The layout of these four words is specified by the file's
+thread**, denoted by its thread ID.  A data set consisting of multiple
+concurrent threads is transmitted or stored as a serial sequence of frames
+called a **data stream**.  The collection of frames that cover all threads for a
+single time segment - equivalently, all thread IDs for the same header time and
+frame number - is a **dataframe set** (or just "frame set").
 
 Strict time ordering of frames in the stream, while considered part of VDIF
 best practices, is not mandated, and cannot be guaranteed during data
 transmission over the internet.
 
 .. _vdif_usage:
 
-Usage
-=====
+Usage Notes
+===========
 
-The examples below use the small sample VDIF file :file:`baseband/data/sample.vdif`,
-and assumes the following modules have been imported::
+This section covers VDIF-specific features of Baseband.  Tutorials for general
+usage can be found under the :ref:`Using Baseband <using_baseband_toc>` section.
+The examples below use the small sample file ``baseband/data/sample.vdif``,
+and assume the `numpy` and `baseband.vdif` modules have been imported::
 
     >>> import numpy as np
     >>> from baseband import vdif
+    >>> from baseband.data import SAMPLE_VDIF
 
-Simple reading and writing of VDIF files can be done entirely using
-:func:`~baseband.vdif.open`, the master input/output function.
-
-Opening in binary mode provides a normal file reader but extended with
-methods to read a :class:`~baseband.vdif.VDIFFrame` (or
-:class:`~baseband.vdif.FrameSet`)::
+Baseband defines the :class:`~baseband.vdif.VDIFFrameSet` data container for
+storing a frame set as well as :class:`~baseband.vdif.VDIFFrame` one for storing
+a single frame. Opening in a VDIF file binary mode provides a file reader
+extended with methods to read both::
 
-    >>> from baseband.data import SAMPLE_VDIF
     >>> fh = vdif.open(SAMPLE_VDIF, 'rb')
     >>> fs = fh.read_frameset()
     >>> fs.data.shape
     (8, 20000, 1)
+    >>> fr = fh.read_frame()
+    >>> fr.data.shape
+    (20000, 1)
     >>> fh.close()
 
-Opening in stream mode wraps the low-level routines such that reading
-and writing is in units of samples.  It also provides access to header
-information::
-
-    >>> fh = vdif.open(SAMPLE_VDIF, 'rs')
-    >>> fh
-    <VDIFStreamReader name=... offset=0
-        nthread=8, samples_per_frame=20000, nchan=1,
-        frames_per_second=1600, complex_data=False, bps=2, edv=3,
-        station=65532, (start) time=2014-06-16T05:56:07.000000000>
-    >>> d = fh.read(12)
-    >>> d.shape
-    (12, 8)
-    >>> d[:, 0].astype(int)  # first thread
-    array([-1, -1,  3, -1,  1, -1,  3, -1,  1,  3, -1,  1])
-    >>> fh.close()
+As with other formats, ``fr.data`` is a read-only property of the frame. 
+``fs.data``, though, is a lazy property which, when it is first called, decodes
+the entire frame set payload into a `numpy.ndarray`.  The values in ``fs.data``
+can freely be modified, but are *not* transmitted back to the raw payload data.
```

Comment:
This sounds like a bug. My tendency would be to omit this for now.

Reply:
Omitted.